lstchain/reco/dl1_to_dl2.py

```diff
@@ -794,11 +806,11 @@ def get_expected_source_pos(data, data_type, config, effective_focal_length=29.3
     # For proton MC, nominal source position is one written in config file
     if data_type == 'mc_proton':
         expected_src_pos = utils.sky_to_camera(
-            u.Quantity(data['mc_alt_tel'].values + config['mc_nominal_source_x_deg'], u.deg, copy=False),
-            u.Quantity(data['mc_az_tel'].values + config['mc_nominal_source_y_deg'], u.deg, copy=False),
+            u.Quantity(data['mc_alt_tel'].values + np.deg2rad(config['mc_nominal_source_x_deg']), u.rad, copy=False),
```

Comment:
I am confused about this.... you add something called "nominal source_{x,y}" to something called az/alt?

This seems wrong, at least if the operation is correct, than the names.

Reply:
This parameter is already there, it specifies the relative position (0.4 deg on x-axis by default) to compute expected source position for proton MC. 
https://github.com/cta-observatory/cta-lstchain/blob/c9e4c4f2d1d44ba1066be86865b770613d6cca1a/lstchain/data/lstchain_src_dep_config.json#L62-L63

Just I found the bug, though it didn't affect the final result much

Reply:
> (0.4 deg on x-axis by default)

What x-axis? In which coordinate system?

Reply:
yep, indeed it may be confusing... It assumes in AltAz coordinates (0.4 deg).. This parameter was introduced here:
https://github.com/cta-observatory/cta-lstchain/pull/247#issuecomment-605840586

Reply:
> 
> yep, indeed it may be confusing... It assumes in AltAz coordinates (0.4 deg).. This parameter was introduced here: [#247 (comment)](https://github.com/cta-observatory/cta-lstchain/pull/247#issuecomment-605840586)
            
u.Quantity(data['mc_az_tel'].values + np.deg2rad(config['mc_nominal_source_y_deg']), u.rad, copy=False),

It is not correct, camera frame's x is "along the altitude axis", but  y  is definitely not the azimuth axis.... 

Also, why do you need this? mc_nominal_source_x_deg and mc_nominal_source_y_deg are already the position on the camera (well, in degrees and without aberration) w.r.t. which you want the src-dep reco for protons.
You just have to transform it to meters

expected_src_pos_y_m = np.tan(np.deg2rad(mc_nominal_source_y_deg)) * effective_focal 
expected_src_pos_x_m = np.tan(np.deg2rad(mc_nominal_source_x_deg)) * effective_focal

Reply:
yep, actually I don't remember why I implemented like this...
I implemented it as Abelardo suggested.

Reply:
I just resolved the conflict resulting from the azimuth range change

Reply:
> expected_src_pos_y_m = np.tan(np.deg2rad(mc_nominal_source_y_deg)) * effective_focal_length
> expected_src_pos_x_m = np.tan(np.deg2rad(mc_nominal_source_x_deg)) * effective_focal_length

We have the ctapipe coordinate systems for this.

The source position is given in AltAz, correct? or where does this `mc_nominal_source_{x,y}_deg` come from?

Then:

```
source_pos = SkyCoord(alt=..., az=.., frame=AltAz())
pointing = SkyCoord(alt=..., az=..., frame=AltAz())
camera_frame = CameraFrame(telescope_pointing=pointing, focal_length=effective_focal_length)

source_camera = source_pos.transform_to(camera_frame)
```

Reply:
No, the "fake position" for the proton analysis is provided in "camera coordinates" x, y, but in degrees and without aberration. So this converts it to true camera coordinates.

Reply:
Ok, then  it's:

```
source_pos = NominalFrame(fov_lon=..., fov_lat=.., frame=TelescopeFrame())
camera_frame = CameraFrame(focal_length=effective_focal_length)
source_camera = source_pos.transform_to(camera_frame)
```

Reply:
> Ok, then it's:
> 
> ```
> source_pos = SkyCoord(fov_lon=..., fov_lat=.., frame=TelescopeFrame())
> camera_frame = CameraFrame(focal_length=effective_focal_length)
> source_camera = source_pos.transform_to(camera_frame)
> ```

Yes, and I think  fov_lon = -mc_nominal_source_y_deg  and  fov_lat = mc_nominal_source_x_deg   

It does not matter much, however, because the direction of the "nominal" position chosen for protons is not critical I think.